codes/kotlin/chapter_computational_complexity/iteration/iteration.kt

```diff
@@ -0,0 +1,74 @@
+/**
+ * File: iteration.kt
+ * Created Time: 2024-01-25
+ * Author: curtishd (1023632660@qq.com)
+ */
+
+package chapter_computational_complexity.iteration
```

Comment:
Can we remove the `iteration` folder to make the relative path consistent with it in other programming languages?

`codes/kotlin/chapter_computational_complexity/iteration/iteration.kt` -> `codes/kotlin/chapter_computational_complexity/iteration.kt`

Reply:
Great! Could you make the same modification on the backtracking chapter? Thx

Reply:
But I am not sure which part of chapter backtracking should be modified.

Reply:
> Can we remove the `iteration` folder to make the relative path consistent with it in other programming languages?
> 
> `codes/kotlin/chapter_computational_complexity/iteration/iteration.kt` -> `codes/kotlin/chapter_computational_complexity/iteration.kt`

Move the files out of the sub-folders. Just like what you did to this chapter's code files

Reply:
You can only remove the file out of the subfolder while keeping the package name different:

- Use package name: `package chapter_computational_complexity.recursion`
- File path: `kotlin/chapter_computational_complexity/recursion.kt`

Kotlin does not require the package name to be equal to the file path, although it's an industry standard.

Reply:
i see.

Reply:
I am sorry did i miss something?

Reply:
> I am sorry did i miss something?

The package name should be `chapter_computational_complexity.iteration`

You need to make the same modification(move them out of the subfolders) to all the code files in the computational complexity chapter and backtracking chapter.

Please carefully check all the changes and the latest file status in https://github.com/krahets/hello-algo/pull/1090/files

Reply:
I make all changes this time. Please take a look.

Reply:
Great! Have you tested all the code and ensure they're runnable?

Reply:
Yes, the code of all files are work well.